Classes/Service/SessionHandler.php

```diff
@@ -74,11 +81,17 @@ public function restoreAddress(string $key): ?AbstractAddress
     /**
      * writes an AbstractAddress object to session
      */
-    public function writeAddress(string $key, AbstractAddress $address)
+    public function writeAddress(string $key, AbstractAddress $address): void
     {
         $sessionData = serialize($address);
 
-        $GLOBALS['TSFE']->fe_user->setKey('ses', $this->prefixKey . $key, $sessionData);
-        $GLOBALS['TSFE']->fe_user->storeSessionData();
+        $this->feUser->setKey('ses', $this->prefixKey . $key, $sessionData);
+        $this->feUser->storeSessionData();
+    }
+
+    private function getSession(): void
```

Comment:
The method should be renamed to something more appropriate.

Something like that:
```suggestion
    private function getFrontendUserFromRequest(): void
```

Reply:
Good idea. Done